bitcoinspv/bootstrap.go

```diff
@@ -145,11 +146,35 @@ func (r *Relayer) initializeBTCCache(ctx context.Context) error {
 	// submitting headers from the light clients height - confirmationDepth (usually 6).
 	baseHeight := blockHeight - r.btcConfirmationDepth + 1
 
-	// NOTE: here we are fetching only headers, if we want to fetch full blocks change the flag to true
-	r.logger.Info().Msg("Fetching headers...")
-	blocks, err := r.btcClient.GetBTCTailBlocksByHeight(baseHeight, false)
+	r.logger.Info().Msg("Fetching blocks/headers for cache and Walrus storage...")
+	fetchFullBlocks := r.Config.StoreBlocksInWalrus && r.walrusHandler != nil
+	blocks, err := r.btcClient.GetBTCTailBlocksByHeight(baseHeight, fetchFullBlocks)
 	if err != nil {
-		return fmt.Errorf("failed to get headers: %w", err)
+		return fmt.Errorf("failed to get blocks/headers: %w", err)
+	}
+
+	// Store full blocks in Walrus
+	if fetchFullBlocks {
+		r.logger.Info().Msgf("Attempting to store %d blocks to Walrus", len(blocks))
+		for _, ib := range blocks {
+			if ib.RawMsgBlock != nil {
+				var blockBuffer bytes.Buffer
+				if err := ib.RawMsgBlock.Serialize(&blockBuffer); err != nil {
+					r.logger.Error().Err(err).Msgf(
+						"failed to serialize block %d (%s) for Walrus",
+						ib.BlockHeight, ib.BlockHash().String(),
+					)
+					continue
+				}
+				_, walrusErr := r.walrusHandler.StoreBlock(blockBuffer.Bytes(), ib.BlockHeight, ib.BlockHash().String())
+				if walrusErr != nil {
+					r.logger.Warn().Err(walrusErr).Msgf(
+						"Walrus store failed for block %d (%s), continuing bootstrap.",
+						ib.BlockHeight, ib.BlockHash().String(),
+					)
+				}
+			}
+		}
```

Comment:
let's move to helper function `UploadToWalrus`

Reply:
done